bullet-train-co/bullet_train-core
# Verify that user has setup authenticator codes before enabling 2FA

closes https://github.com/bullet-train-co/bullet_train/issues/539
closes https://github.com/bullet-train-co/bullet_train/issues/538

**Description**
The goal of this PR is to prevent users from locking themselves out of their account while enabling two-factor authentication. We do this by making users enter the Verification Code from their Authenticator app to prove that they have scanned the QR code and installed an Authenticator app on a device they can access.

In order to verify the user entered `otp_code` we use the same method used during regular sign-in device authentication, which is part of the `device-two-factor` gem. 

**Testing Done:**
- Click 'Enable' 2FA
- Scan the QR code and enter the correct Verification Code and click Verify
- see the 'happy path' message in screenshot below. Note that 2FA is enabled in the DB.
- Click 'Disable'
- Click 'Enable' again
- This is type enter a random/incorrect Verification Code
- see the 'sad path' message below. And note that 2FA is not enabled in the DB.

<img width="734" alt="Screen Shot 2023-01-19 at 11 10 46 AM" src="https://user-images.githubusercontent.com/1966473/213515493-005a91f0-464f-4090-85dc-515b86ccb77b.png">
<img width="712" alt="Screen Shot 2023-01-19 at 11 12 02 AM" src="https://user-images.githubusercontent.com/1966473/213515487-cafe7afa-017c-4e37-82ad-594c72f3a3ca.png">
<img width="717" alt="Screen Shot 2023-01-19 at 11 11 10 AM" src="https://user-images.githubusercontent.com/1966473/213515491-8fde564d-0d8e-4fe8-9b2d-0afa2e2ac1c4.png">
